src/main/java/com/alexfu/sqlitequerybuilder/utils/StringUtils.java

```diff
@@ -1,19 +1,19 @@
 package com.alexfu.sqlitequerybuilder.utils;
 
 public class StringUtils {
-  public static String join(String delimeter, String... array) {
-    return join(delimeter, (Object[]) array);
+  public static String join(CharSequence delimiter, String... array) {
+    return join(delimiter, (Object[]) array);
   }
 
-  public static String join(String delimeter, Object... array) {
-    String result = "";
+  public static String join(CharSequence delimiter, Object... array) {
+    StringBuilder sb = new StringBuilder();
+
     for (int i = 0, size = array.length; i < size; i++) {
```

Comment:
For info, the `foreach` loop dissambled code produces exactly this statement for primitive arrays.